packages/gatsby-plugin-mdx/README.md

```diff
@@ -1,128 +1,93 @@
-      - [Imports](#imports)
-      - [Shortcodes](#shortcodes)
-      - [Gatsby remark plugins](#gatsby-remark-plugins)
-      - [Remark plugins](#remark-plugins)
-      - [Rehype plugins](#rehype-plugins)
-      - [Media types](#media-types)
-        - [Explanation](#explanation)
-      - [shouldBlockNodeFromTransformation](#shouldblocknodefromtransformation)
-    - [Components](#components)
-      - [MDXProvider](#mdxprovider)
-        - [Related](#related)
-      - [MDXRenderer](#mdxrenderer)
-  - [License](#license)
+- [Installation](#installation)
+- [Usage](#usage)
+  - [Configuration](#configuration)
+    - [Extensions](#extensions)
+    - [Gatsby remark plugins](#gatsby-remark-plugins)
+    - [mdxOptions](#mdxoptions)
+  - [Imports](#imports)
+  - [Layouts](#layouts)
+  - [Programmatically create MDX pages](#programmatically-create-mdx-pages)
+  - [GraphQL MDX Node structure](#graphql-mdx-node-structure)
+  - [Extending the GraphQL Mdx nodes](#extending-the-graphql-mdx-nodes)
+  - [Components](#components)
+    - [MDXProvider](#mdxprovider)
+    - [Shortcodes](#shortcodes)
+- [Migrating from v3 to v4](#migrating-from-v3-to-v4)
+- [Why MDX?](#why-mdx)
+- [Related](#related)
 
 ## Installation
 
-Install:
-
 ```shell
-npm install gatsby-plugin-mdx @mdx-js/mdx@v1 @mdx-js/react@v1
+npm install gatsby-plugin-mdx @mdx-js/react gatsby-source-filesystem
 ```
 
 ## Usage
 
 After installing `gatsby-plugin-mdx` you can add it to your plugins list in your
-`gatsby-config.js`.
+`gatsby-config.js`. You'll also want to configure `gatsby-source-filesytem` to point at your `src/pages` directory (even if you don't want to create MDX pages from `src/pages`).
 
-```js
+```js:title=gatsby-config.js
 module.exports = {
   plugins: [
+    `gatsby-plugin-mdx`,
     {
       resolve: `gatsby-source-filesystem`,
       options: {
         name: `pages`,
-        path: `${__dirname}/src/pages/`,
+        path: `${__dirname}/src/pages`,
       },
     },
-    `gatsby-plugin-mdx`,
   ],
 }
 ```
 
-By default, this configuration will allow you to automatically create pages
-with `.mdx` files in `src/pages` and will process any Gatsby nodes
-with Markdown media types into MDX content.
+By default, this configuration will allow you to automatically create pages with `.mdx` files in `src/pages` and will process any Gatsby nodes with Markdown media types into MDX content.
 
-Note that `gatsby-plugin-mdx` requires `gatsby-source-filesystem` to be present
-and configured to process local markdown files in order to
-generate the resulting Gatsby nodes.
+Note that `gatsby-plugin-mdx` requires `gatsby-source-filesystem` to be present and configured to process local markdown files in order to generate the resulting Gatsby nodes.
 
-To automatically create pages with `.mdx` from other sources, you also need
-to configure `gatsby-plugin-page-creator`.
+To automatically create pages with `.mdx` from other sources, you also need to configure `gatsby-plugin-page-creator`.
```

Comment:
Would they need to use this instead of an additional `gatsby-source-filesystem`? It doesn't look like there's mention of `gatsby-plugin-page-creator` in the dotcom docs page

Reply:
If you want to have MDX files automatically turned into pages outside of `src/pages` (which is a bit of an uncommon use case) you'll need to configure the `gatsby-source-filesystem` and `gatsby-plugin-page-creator`.

I think "from other sources" is misleading here, it's only relevant for other local sources in another location

Reply:
```suggestion
To automatically create pages from MDX files outside of `src/pages` you'll need to configure `gatsby-plugin-page-creator` and `gatsby-source-filesystem` to point to this folder of files.
```

Reply:
Better like that?

Reply:
Yeah that's clearer to me now, thanks!